### PR TITLE
completed assessment

### PR DESCRIPTION
**Pull request by kevans107**

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -23,10 +24,31 @@ var secretCodeWord2 = "Gobbledygook"
...
+    finalString = ''
+    // Create an empty string to contain the new string since .replace does not alter the original string
+    for(let i=0; i < string.length; i++){
+        // Iterate through the string with a for loop
+        finalString = string.replace(/([aA])/g, "4").replace(/([eE])/g, '3').replace(/([iI])/g, '1').replace(/[oO]/g, '0')
+        }
+        // Use the .replace method with regex to replace each vowel with the corresponding number. Perform a global replacement to replace all instances of the vowel.
+        return finalString
+    }
+
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:50]: Alright this works! But it doesn't need a for loop. For loops give you access to the value if "i". 

You aren't calling on "i" and you have a return in your for loop that stops the for loop. So here is your code just refactored to look like what it is already doing.
```javascript 
// Create a function called codedMessage  that takes in a string as a parameter
const codedMessage = (string) => {
    // Create an empty string to contain the new string since .replace does not alter the original string
    finalString = ''
        // Use the .replace method with regex to replace each vowel with the corresponding number. Perform a global replacement to replace all instances of the vowel.
   finalString = string.replace(/([aA])/g, "4").replace(/([eE])/g, '3').replace(/([iI])/g, '1').replace(/[oO]/g, '0')
        
      
   return finalString
}
```

File: code-challenges.test.js [hunk 2]
```diff
@@ -57,6 +98,34 @@ var hand2 = [5, 5, 3, 3, 4]
...
+        // Us for...in to count the number of instances for each value in the object
+        houseArray.push(counts[key])
+        // push the key and the counts for each value to the empty houseArray
+    }
+    for(let i=0;i<houseArray.length;i++){
+      if(houseArray[i]===3)return true
+      // Use for loop to iterate through the houseArray, if one value is only returned three times, return true
+    }return false
+    // else return false
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:131]: You did a great job at explaining the process of logic here. I did find an edge case that broke the logic 
```
> console.log(fullHouse([5,5,5,2,1]))
> true
```

I think this can be fixed by adding an extra conditional after line 127.

File: interview-questions.md [hunk 3]
```diff
@@ -7,53 +7,53 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. Are (most) Ruby methods accessors or mutators?
 
-  Your answer:
+  Your answer: Most Ruby methods are accessors, rather than mutators, meaning they access information but do not permanently change the data. In cases where a developer wants to permanently alter the information a bang operator can be added to the accessor after the dot method is used. For example .reverse mutates the information where as .reverse! makes the reverse permanent.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:10]: Perfect answer!

File: interview-questions.md [hunk 4]
```diff
@@ -7,53 +7,53 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  Researched answer: A gem is an open source library that contains Ruby code and are packaged to add functionality for developers. Gems allow developers to use code snippets to improve functionality without having to take the time to write all the lines of code to do the same processes. Gems help to improve efficiency and reduce redundant code by using pre-existing gems. One example of a gem is the Rspec gem, which is a testing framework that helps developers test Ruby code.
 
 
 
 4. What is a relational database? Are there other kinds of databases?
 
-  Your answer:
+  Your answer: A relational database is a database with rows and tables of related content. In a relational database each instance in the database is assigned a primary key to identify each row as a unique object within the database. Each object is related to each other within the database. There are other kinds of databases that I would have to do more research on to give a more thorough answer.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:34]: This is a really good start

File: interview-questions.md [hunk 5]
```diff
@@ -7,53 +7,53 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 5. What are primary keys? Why are they important?
 
-  Your answer:
+  Your answer: Primary keys are a unique id given to every unique instance in a relational database. A primary key is especially important in databases where there may be more than one entry with the same information. In these cases a primary key will be able to tell that each row is a unique instance. For example, in a database that has inputs for people there may be a father and son who have the same name and live at the same address. A primary key will be able to identify each row of information as a unique instance and potentially avoid confusion that they are duplicates.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:42]: YES!

File: ruby-challenges.rb [hunk 6]
```diff
@@ -10,6 +10,17 @@
 letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
 
+def single_letter(array, value)
+    # Create a method called single_letter
+        array.select { |string| string.include?(value)}
+    # Use the .select to iterate through the array
+    # Inside the block use the .include? to return a boolean of whether the string includes a specific character
+    end
```
Comment (WalkerTexas-coder) [inline ruby-challenges.rb:18]: Excellent thank you for pseudo code explaining what's going on!

File: ruby-challenges.rb [hunk 7]
```diff
@@ -19,16 +30,65 @@
 nums_array2 = [25, 17, 47, 11]
 # Expected output: 100
 
+def return_sum array
+    # create a method called return_sum that has an array as a parameter
+    array.inject { |sum, num| sum + num}
```
Comment (WalkerTexas-coder) [inline ruby-challenges.rb:35]: Oh that's way cool

File: ruby-challenges.rb [hunk 8]
```diff
@@ -19,16 +30,65 @@
...
+        @brake = 0
+    end
+    def bike_info
+        p "The Trek bike has #{@wheels} wheels and is going #{@current_speed} mph."
+    end
+    def pedal_faster(pedal_faster)
+        p @current_speed = @current_speed + @pedal_faster
+    end
+    def pedal_brake(brake)
+        if @current_speed >= 0
```
Comment (WalkerTexas-coder) [inline ruby-challenges.rb:84]: There's a conditional statement addition here that would make this class a bit more robust. 

```ruby
def pedal_brake(brake)
        if @current_speed - brake > 0
           p @current_speed = @current_speed - @brake
        else 
           @current_speed = 0
end
```

**Review by WalkerTexas-coder — APPROVED** [review]
Kelly, good work!You have taken some big steps in the right direction and I can see you exploring new ideas! The two stand out points of this assessment are your interview questions and pseudo code/logic, excellent stuff!  I left more specific comments below for your reference. This PR is approved!
